src/Playwright.Tests/BrowserTypeConnectTests.cs

```diff
@@ -80,71 +82,396 @@ public void TearDown()
+            {
+                context.Response.Headers["Content-Type"] = "application/octet-stream";
+                context.Response.Headers["Content-Disposition"] = "attachment;";
+                return context.Response.WriteAsync("Hello world");
+            });
+
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var page = await browser.NewPageAsync(new() { AcceptDownloads = true });
+            await page.SetContentAsync($"<a href=\"{Server.Prefix}/download\">download</a>");
+            var downloadTask = page.WaitForDownloadAsync();
+
+            await TaskUtils.WhenAll(
+                downloadTask,
+                page.ClickAsync("a"));
+
+            using var tmpDir = new TempDirectory();
+            string userPath = Path.Combine(tmpDir.Path, "these", "are", "directories", "download.txt");
+            var download = downloadTask.Result;
+            await download.SaveAsAsync(userPath);
+            Assert.True(new FileInfo(userPath).Exists);
+            Assert.AreEqual("Hello world", File.ReadAllText(userPath));
+            await page.CloseAsync();
+            var exception = await PlaywrightAssert.ThrowsAsync<PlaywrightException>(() => download.PathAsync());
+            StringAssert.Contains("Target page, context or browser has been closed", exception.Message);
         }
 
-        [PlaywrightTest("browsertype-connect.spec.ts", "should respect selectors")]
-        [Ignore("SKIP WIRE")]
-        public void ShouldRespectSelectors()
+        [PlaywrightTest("browsertype-connect.spec.ts", "should error when saving download after deletion")]
+        public async Task ShouldErrorWhenSavingDownloadAfterDeletion()
         {
+            Server.SetRoute("/download", context =>
+            {
+                context.Response.Headers["Content-Type"] = "application/octet-stream";
+                context.Response.Headers["Content-Disposition"] = "attachment; filename=file.txt";
+                return context.Response.WriteAsync("Hello world");
+            });
+
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var page = await browser.NewPageAsync(new() { AcceptDownloads = true });
+            await page.SetContentAsync($"<a href=\"{Server.Prefix}/download\">download</a>");
+            var downloadTask = page.WaitForDownloadAsync();
+
+            await TaskUtils.WhenAll(
+                downloadTask,
+                page.ClickAsync("a"));
+
+            using var tmpDir = new TempDirectory();
+            string userPath = Path.Combine(tmpDir.Path, "download.txt");
+            var download = downloadTask.Result;
+            await download.DeleteAsync();
+            var exception = await PlaywrightAssert.ThrowsAsync<PlaywrightException>(() => download.SaveAsAsync(userPath));
+            StringAssert.Contains("Target page, context or browser has been closed", exception.Message);
+        }
+
```

Comment:
"should properly disconnect when connection closes from the client side" test is missing
"should properly disconnect when connection closes from the server side" test is missing

Reply:
As discussed above, need to update the SimpleServer for these tests. Will create a separate PR for modifying "SimpleServer.cs" and writing the tests.